File: shoring/blackjackEngine/src/main/java/com/panopset/blackjackEngine/CountingSystem.java
package com.panopset.blackjackEngine;

import java.util.HashMap;
import java.util.Map;
import java.util.StringTokenizer;
import com.panopset.compat.Stringop;

public class CountingSystem implements Comparable<CountingSystem> {

  private int count = 0;
  private final Integer sel;
  private final String nm;
  private final Map<String, Integer> vals = new HashMap<>();

  public CountingSystem(final String name, final String dta, final int selection) {
    sel = selection;
    nm = name;
    StringTokenizer st = new StringTokenizer(dta, " ");
    vals.put("2", Stringop.parseInt(st.nextToken().trim()));
    vals.put("3", Stringop.parseInt(st.nextToken().trim()));
    vals.put("4", Stringop.parseInt(st.nextToken().trim()));
    vals.put("5", Stringop.parseInt(st.nextToken().trim()));
    vals.put("6", Stringop.parseInt(st.nextToken().trim()));
    vals.put("7", Stringop.parseInt(st.nextToken().trim()));
    vals.put("8", Stringop.parseInt(st.nextToken().trim()));
    vals.put("9", Stringop.parseInt(st.nextToken().trim()));
    vals.put("T", Stringop.parseInt(st.nextToken().trim()));
    vals.put("A", Stringop.parseInt(st.nextToken().trim()));
  }

  public String getName() {
    return nm;
  }

  public int getCount() {
    return count;
  }

  public void count(BlackjackCard blackjackCard) {
    if (blackjackCard.isAce()) {
      count = count + vals.get("A");
    } else {
      String valRep = String.format("%d", blackjackCard.getSoftValue());
      if ("10".equals(valRep)) {
        valRep = "T";
      }
      count = count + vals.get(valRep);
    }
  }

  public Integer getSelection() {
    return sel;
  }

  @Override
  public int compareTo(final CountingSystem o) {
    return getSelection().compareTo(o.getSelection());
  }

  @Override
  public boolean equals(final Object o) {
    return (o instanceof CountingSystem)
        && getSelection().equals(((CountingSystem) o).getSelection());
  }

  @Override
  public int hashCode() {
    return getSelection().hashCode();
  }

  public void resetCount() {
    count = 0;
  }
}

File: code/shoring/blackjackEngine/src/main/kotlin/com/panopset/blackjackEngine/CountingSystem.kt
package com.panopset.blackjackEngine

import com.panopset.compat.Logz
import com.panopset.compat.Stringop
import java.util.*

class CountingSystem(val name: String, dta: String, val selection: Int) : Comparable<CountingSystem> {
    var count = 0
        private set
    private val vals: MutableMap<String, Int> = HashMap()

    init {
        val st = StringTokenizer(dta, " ")
        vals["2"] = Stringop.parseInt( st.nextToken().trim())
        vals["3"] = Stringop.parseInt( st.nextToken().trim())
        vals["4"] = Stringop.parseInt( st.nextToken().trim())
        vals["5"] = Stringop.parseInt( st.nextToken().trim())
        vals["6"] = Stringop.parseInt( st.nextToken().trim())
        vals["7"] = Stringop.parseInt( st.nextToken().trim())
        vals["8"] = Stringop.parseInt( st.nextToken().trim())
        vals["9"] = Stringop.parseInt( st.nextToken().trim())
        vals["T"] = Stringop.parseInt( st.nextToken().trim())
        vals["A"] = Stringop.parseInt( st.nextToken().trim())
    }

    fun count(blackjackCard: BlackjackCard) {
        if (blackjackCard.isAce) {
            count += vals["A"]!!
        } else {
            var valRep = String.format("%d", blackjackCard.softValue)
            if ("10" == valRep) {
                valRep = "T"
            }
            count += vals[valRep]!!
        }
    }

    override fun compareTo(other: CountingSystem): Int {
        return selection.compareTo(other.selection)
    }

    override fun equals(other: Any?): Boolean {
        return other is CountingSystem && selection == other.selection
    }

    override fun hashCode(): Int {
        return selection.hashCode()
    }

    fun resetCount() {
        count = 0
    }
}
